pkg/wireguard/agent/agent.go

```diff
@@ -88,30 +93,113 @@ func NewAgent(privKeyPath string) (*Agent, error) {
 		peerByNodeName:   map[string]*peerConfig{},
 		nodeNameByNodeIP: map[string]string{},
 		restoredPubKeys:  map[wgtypes.Key]struct{}{},
+		cleanup:          []func(){},
 	}, nil
 }
 
 // Close is called when the agent stops
 func (a *Agent) Close() error {
+	a.Lock()
```

Comment:
Could this be `a.RLock` given `a.cleanup` is not modified?

Reply:
Addressed!